### ISSUE TITLE
Plugin working against some Windows Auth sites, but not others

### ISSUE THREAD

**Issue by joshuakeel** [bjowes/cypress-ntlm-auth#23]
Hi there,

Thanks for making this great plugin. I was able to quickly get it set up and working!

My only issue is that I can get it to work against some of our sites, but not all. I have checked for configuration differences, and the Authentication settings in IIS for both are set up the same way. However when I try to visit one of the sites, I get a 401 Unauthorized. It seems like somehow the negotiation is failing or something.

I tried running the proxy in debug mode, but it didn't output much that seemed useful:

```
  cypress:plugin:ntlm-auth NTLM auth proxy listening on port: 50733 +0ms
  cypress:plugin:ntlm-auth NTLM auth config API listening on port: 50734 +6ms
  cypress:plugin:ntlm-auth wrote C:\Users\username\AppData\Roaming\cypress-ntlm-auth\cypress-ntlm-auth.port +0ms
  cypress:plugin:ntlm-auth Startup done! +0ms
  cypress:plugin:ntlm-auth { ntlmProxyUrl: 'http://127.0.0.1:50733',
  cypress:plugin:ntlm-auth   configApiUrl: 'http://127.0.0.1:50734' } +1ms
  cypress:plugin:ntlm-auth Tunnel to download.cypress.io:443 +16s
  cypress:plugin:ntlm-auth Got ECONNRESET on PROXY_TO_SERVER_SOCKET, ignoring. +217ms
  cypress:plugin:ntlm-auth Request to http://127.0.0.1:50734/ - pass on +33s
  cypress:plugin:ntlm-auth Created non-NTLM agent for target http://127.0.0.1:50734/ +1ms
  cypress:plugin:ntlm-auth Received reset +24ms
  cypress:plugin:ntlm-auth Removed all agents due to reset +1ms
  cypress:plugin:ntlm-auth Request to http://ktsqnwsejbq/ - pass on +2s
  cypress:plugin:ntlm-auth Created non-NTLM agent for target http://ktsqnwsejbq/ +1ms
  cypress:plugin:ntlm-auth Request to http://vfdytfevn/ - pass on +4ms
  cypress:plugin:ntlm-auth Created non-NTLM agent for target http://vfdytfevn/ +2ms
  cypress:plugin:ntlm-auth Request to http://fhcddevflxywqo/ - pass on +3ms
  cypress:plugin:ntlm-auth Created non-NTLM agent for target http://fhcddevflxywqo/ +1ms
  cypress:plugin:ntlm-auth Chrome startup HEAD request detected (host: ktsqnwsejbq). Ignoring connection error. +23ms
  cypress:plugin:ntlm-auth Chrome startup HEAD request detected (host: vfdytfevn). Ignoring connection error. +4ms
  cypress:plugin:ntlm-auth Chrome startup HEAD request detected (host: fhcddevflxywqo). Ignoring connection error. +22ms
  cypress:plugin:ntlm-auth Request to http://127.0.0.1:50734/ - pass on +2m
  cypress:plugin:ntlm-auth Created non-NTLM agent for target http://127.0.0.1:50734/ +2ms
  cypress:plugin:ntlm-auth Received reset +2ms
  cypress:plugin:ntlm-auth Removed all agents due to reset +1ms
  cypress:plugin:ntlm-auth Request to http://127.0.0.1:50734/ - pass on +24ms
  cypress:plugin:ntlm-auth Created non-NTLM agent for target http://127.0.0.1:50734/ +1ms
  cypress:plugin:ntlm-auth Received valid config update +2ms
  cypress:plugin:ntlm-auth Tunnel to my-domain.com:443 +35ms
  cypress:plugin:ntlm-auth Tunnel to my-domain.com:443 +62ms
```

Any idea why one site would work, and the other not? These sites are set up on the same server, and should essentially be configured exactly the same (aside from the fact that they're running different applications). The only thing I can figure is that the application is returning a 401 because of some special requirement it has.

**Comment by bjowes** [comment]
Hi @joshuakeel , I'm glad you like it!
The log you submitted shows no configuration of NTLM hosts, as if the cy.ntlm command is never called. In that case it will not handle authentication for any sites. Are you sure this is the ntlm-proxy instance being used?
Also, the last two lines are a hint - "Tunnel to my-domain.com:443" means that there was an incoming https request for a site which isn't registered for NTLM, hence the proxy sets up a tunnel for the site (for performance reasons). If this is the site you are testing, this is where your traffic currently is.

**Comment by joshuakeel** [comment]
Thanks, @bjowes. I tried again, verifying everything along the way to be sure I set everything up right. I just tried again with a slightly different test setup, and I get this:

```
  cypress:plugin:ntlm-auth Request to https://my-domain.com/ - pass on +37ms
  cypress:plugin:ntlm-auth Created non-NTLM agent for target https://my-domain.com/ +1ms
  cypress:plugin:ntlm-auth Request to https:/my-domain.com/ - pass on +60ms
  cypress:plugin:ntlm-auth Created non-NTLM agent for target https://my-domain.com/ +1ms
```

my-domain.com represents the domain I'm testing with (although of course it isn't the actual domain).

My test setup looks like this:

```
describe('some test', () => {

  beforeEach(() => {
    cy.ntlmReset();
  });

  it('should load NTLM site', () => {

    cy.ntlm('https://my-domain.com:443', '[username]', '[password]', '[domain]');
    cy.visit('https://my-domain.com/');
  });
});
```

If I leave off the port 443 from the hostname in the `cy.ntlm` call, I get the original results I pasted in my first message. So I'm calling to register the site, but it doesn't wind up registered.

**Comment by joshuakeel** [comment]
I just discovered something else that is kind of odd. When I tried another of our sites, I can get it working, but only over HTTP, not HTTPS. The site that isn't working only allows HTTPS, so maybe that has something to do with it?

**Comment by bjowes** [comment]
Good catch @joshuakeel! There was an issue with NTLM hosts registered without port numbers. It is resolved in version 0.9.2, coming soon.

**Comment by joshuakeel** [comment]
Thanks, @bjowes! It works great for me now! 💯
